packages/components/src/toggle-group-control/toggle-group-control/utils.ts

```diff
@@ -21,24 +21,28 @@ type ValueProp = ToggleGroupControlProps[ 'value' ];
 export function useComputeControlledOrUncontrolledValue(
 	valueProp: ValueProp
 ): { value: ValueProp; defaultValue: ValueProp } {
-	const hasEverBeenUsedInControlledMode = useRef( false );
+	const [
+		hasEverBeenUsedInControlledMode,
+		setHasEverBeenUsedInControlledMode,
+	] = useState( false );
 	const previousValueProp = usePrevious( valueProp );
 
 	useEffect( () => {
-		if ( ! hasEverBeenUsedInControlledMode.current ) {
+		if ( ! hasEverBeenUsedInControlledMode ) {
 			// Assume the component is being used in controlled mode if:
 			// - the `value` prop is not `undefined`
 			// - the `value` prop was not previously `undefined` and was given a new value
-			hasEverBeenUsedInControlledMode.current =
+			setHasEverBeenUsedInControlledMode(
 				valueProp !== undefined &&
-				previousValueProp !== undefined &&
-				valueProp !== previousValueProp;
+					previousValueProp !== undefined &&
+					valueProp !== previousValueProp
+			);
```

Comment:
> That prevents scheduling setState calls that will be noops anyway, changing false to false.

That's a good suggestion, thank you!

> Also, once in controlled mode, the component can't change back to uncontrolled, right?

Yes, this custom behaviour was necessary to be backwards compatible with how the component used to work before supporting uncontrolled mode.

I'm not going to implement this suggestion since I decided to refactor the hook following your other suggestion